### ISSUE TITLE
Installation issues

### ISSUE THREAD

**Issue by avtaniket** [awslabs/logstash-input-dynamodb#17]
Hi,

During setup I can across below issues, using logstash 1.5

1> `jruby -S gem install --local logstash-input-dynamodb-1.0.0-java.gem`
ERROR:  Could not find a valid gem 'logstash-core' (< 2.0.0, >= 1.4.0) in any repository

2>  `sudo  bin/plugin install –no-verify logstash-input-dynamodb`
Validating –no-verify
Plugin –no-verify does not exist
ERROR: Installation aborted, verification failed for –no-verify 

3> `sudo  bin/plugin install –no-verify logstash-input-dynamodb-1.0.0.gem`
ERROR: Mixed source of plugins, you can't mix local `.gem` and remote gems

See: 'bin/plugin install --help'

**Comment by ghost** [comment]
having the same issue, I tried installing logstash-core and picked an arbitrary version that fell in the <2.0.0, >=1.4.0 category (i.e. 1.5.6) and that still is not working. Here's what I'm getting now:

ubuntu@ip-172-31-15-36:~/logstash-input-dynamodb$ jruby -S gem install --local logstash-input-dynamodb-1.0.0-java.gem
ERROR:  While executing gem ... (Gem::DependencyResolutionError)
    conflicting dependencies logstash-core (< 3.0.0, >= 2.0.0.beta2) and logstash-core (< 2.0.0, >= 1.4.0)
  Activated logstash-core-1.5.6
  which does not match conflicting dependency (< 3.0.0, >= 2.0.0.beta2)

  Conflicting dependency chains:
    logstash-input-dynamodb (= 1.0.0), 1.0.0 activated, depends on
    logstash-core (< 2.0.0, >= 1.4.0), 1.5.6 activated

  versus:
    logstash-input-dynamodb (= 1.0.0), 1.0.0 activated, depends on
    logstash-codec-json (>= 0), 2.1.0 activated, depends on
    logstash-core (< 3.0.0, >= 2.0.0.beta2)

  Gems matching logstash-core (< 3.0.0, >= 2.0.0.beta2):
    logstash-core-2.0.0.rc1-java, logstash-core-2.2.2-java, logstash-core-2.3.0.snapshot5-java

I'm not quite sure how dependencies are getting resolved but there's clearly some issues with the way dependencies are defined, that'd be my guess anyway.

**Comment by devnullnor** [comment]
I have the same problem. If this isn't solved soon I'm turning to DocumentDB from MS.

**Comment by marcosnils** [comment]
@devnullnor can't you use the docker container?

**Comment by devnullnor** [comment]
How can I use docker for this task? Will it cost me anything? Thank you.

**Comment by avtaniket** [comment]
@devnullnor Docker is free, I tried with docker and now its working perfectly.

Check link https://hub.docker.com/r/mantika/logstash-dynamodb-streams/

1> Step 1 - Install docker if you don't have already

2> Step 2 - Pull image 
`$ docker pull mantika/logstash-dynamodb-streams`

Hope this help you to get started.

**Comment by devnullnor** [comment]
Docker seems to have fixed it. Thank you! Question: Do I need to run
`docker run mantika/logstash-dynamodb-streams -e`
periodically, or is this a continuous stream that will pull data from Dynamo?

**Comment by marcosnils** [comment]
@devnullnor

No, logstash will automatically propagate the changes to elasticsearch, you don't need to run it periodically.

**Comment by devnullnor** [comment]
I get an error:
`{:timestamp=>"2016-04-07T09:59:15.139000+0000", :message=>"Connection refused", :class=>"Manticore::SocketException", :level=>:error}
The error reported is:
  Requested resource not found: Table: users not found (Service: AmazonDynamoDBv2; Status Code: 400; Error Code: ResourceNotFoundException; Request ID: 81F1UHJKBPLIP8GC630S19274JVV4KQNSO5AEMVJF66Q9ASUAAJG)`

Which connection is refused?

**Comment by shikasta-kashti** [comment]
I'm having the same issue on Windows 7. Any updates?

> C:\logstash-input-dynamodb (master)
> λ jruby -S gem install --local logstash-input-dynamodb-1.0.0-java.gem
> ERROR:  Could not find a valid gem 'logstash-core' (< 2.0.0, >= 1.4.0) in any repository

**Comment by marcosnils** [comment]
@mwahidje https://github.com/awslabs/logstash-input-dynamodb/issues/21

**Comment by shikasta-kashti** [comment]
Thanks. It seems to be working now.

**Comment by vamcyleonardo** [comment]
Struggling with the same issue.
Could not find a valid gem 'logstash-core' (< 2.0.0, >= 1.4.0) in any repository.
How to get it to work?

Thanks
